### PR TITLE
Feature/split function arguement

### PR DESCRIPTION
**Pull request by arkadeep16121993**
Added the split function in ClassificationTree and tweaked the split function codes accordingly

### CODE REVIEW

**Comment by arkadeep16121993** [comment]
The test files needs to be changed to pass the checks

File: src/ClassificationTree.jl [hunk 1]
```diff
@@ -1,29 +1,11 @@
 using StatsBase: mode
 
-"""
-Represents a Leaf in the ClassificationTree structure.
```
Comment (cedric-len) [inline src/ClassificationTree.jl:4]: These docstrings should not be deleted!

File: src/ClassificationTree.jl [hunk 2]
```diff
@@ -1,29 +1,11 @@
...
-
-`values` stores the labels of the data points.
-"""
+# Represents a Leaf in the ClassificationTree structure.
 mutable struct Leaf{L}
     values::Vector{L} # represents the prediction/s
 end
 
-"""
-Represents a Node in the ClassificationTree structure.
```
Comment (cedric-len) [inline src/ClassificationTree.jl:13]: see above

File: src/ClassificationTree.jl [hunk 3]
```diff
@@ -37,64 +19,40 @@ mutable struct Node{T, L}
     Node(data::Matrix{T}, labels::Vector{L}) where {T, L} = new{T, L}(missing, missing, -1, missing, data, labels)
 end
 
-"""
-Represents a ClassificationTree.
-
-`max_depth` controls the maximum depth of the tree. If -1, the depth is not limited.
```
Comment (cedric-len) [inline src/ClassificationTree.jl:43]: see above

File: src/ClassificationTree.jl [hunk 4]
```diff
@@ -37,64 +19,40 @@ mutable struct Node{T, L}
...
+    ClassificationTree(max_depth::Int, min_samples_split::Int, split_criterion::Function, data::Matrix{T}, labels::Vector{L}) where {T, L} = 
+        size(data, 1) == length(labels) ? new{T, L}(max_depth, min_samples_split, split_criterion, missing, data, labels) : throw(ArgumentError("The number of rows in data must match the number of elements in labels"))
 end
 
-ClassificationTree(data, labels) = ClassificationTree(-1, 1, data, labels)
-
+ClassificationTree(data, labels, split_criterion) = ClassificationTree(-1, 1, split_criterion, data, labels)
 
-"""
-    build_tree(data, labels, max_depth, min_samples_split, depth)
```
Comment (cedric-len) [inline src/ClassificationTree.jl:75]: see above

File: src/ClassificationTree.jl [hunk 5]
```diff
@@ -115,32 +73,22 @@ function build_tree(data::Matrix{T}, labels::Vector{L}, max_depth::Int, min_samp
...
-    node.right = build_tree(right_data, right_labels, max_depth, min_samples_split, depth + 1)
+    node.left = build_tree(left_data, left_labels, max_depth, min_samples_split, split_criterion, depth + 1)
+    node.right = build_tree(right_data, right_labels, max_depth, min_samples_split, split_criterion, depth + 1)
 
     # Return the node
     return node
 end
 
-"""
-    fit(tree::ClassificationTree)
```
Comment (cedric-len) [inline src/ClassificationTree.jl:126]: see above

File: src/ClassificationTree.jl [hunk 6]
```diff
@@ -164,18 +112,14 @@ function predict(tree::ClassificationTree, data::Matrix{T}) where {T}
     return predictions
 end
 
-"""
-    print_tree(tree:ClassificationTree)
```
Comment (cedric-len) [inline src/ClassificationTree.jl:168]: see above

File: src/SplitCriterion.jl [hunk 7]
```diff
@@ -0,0 +1,19 @@
+module SplitCriterion
```
Comment (cedric-len) [inline src/SplitCriterion.jl:1]: remove this entirely, since it is not used

File: test/test_classificationtree.jl [hunk 8]
```diff
@@ -1,35 +1,40 @@
...
 
 @testset "ClassificationTree" begin
+    # Test 1: Categorical and numerical data with labels
     data = ["dog" 37.0; "dog" 38.4; "dog" 40.2; "dog" 38.9; "human" 36.2; "human" 37.4; "human" 38.8; "human" 36.2]
     labels = ["healthy", "healthy", "sick", "healthy", "healthy", "sick", "sick", "healthy"]
-    # data = ["dog" 37.0; "dog" 38.4; "dog" 40.2; "dog" 38.9; "human" 36.2; "human" 37.4; "human" 38.8; "human" 36.2]
-    # labels = ["healthy", "healthy", "healthy", "healthy", "sick", "sick", "sick", "sick"]
-    tree = ClassificationTree(data, labels)
+    
+    # Using Information Gain criterion
```
Comment (cedric-len) [inline test/test_classificationtree.jl:11]: remove this comment

**Review by cedric-len — CHANGES_REQUESTED** [review]
Please add the docstrings you removed and delete the `SplitCriterion.jl` file

**Pull request closed without merging**